forge-gui/res/cardsfolder/b/borderland_explorer.txt

```diff
@@ -2,7 +2,10 @@ Name:Borderland Explorer
 ManaCost:1 G
 Types:Creature Elf Scout
 PT:3/1
-T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigExploration | TriggerDescription$ When CARDNAME enters the battlefield, each player may discard a card. Each player who discarded a card this way may search their library for a basic land card, reveal it, put it into their hand, then shuffle.
-SVar:TrigExploration:DB$ RepeatEach | RepeatPlayers$ Player | RepeatSubAbility$ DBDiscardToFetch
-SVar:DBDiscardToFetch:DB$ ChangeZone | Origin$ Library | Destination$ Hand | ChangeType$ Land.Basic | DefinedPlayer$ Player.IsRemembered | UnlessCost$ Discard<1/Card> | UnlessPayer$ Player.IsRemembered | UnlessSwitched$ True
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChoose | TriggerDescription$ When CARDNAME enters the battlefield, each player may discard a card. Each player who discarded a card this way may search their library for a basic land card, reveal it, put it into their hand, then shuffle.
+SVar:TrigChoose:DB$ GenericChoice | ShowChoice$ ExceptSelf | ResolveAfterChoosing$ True | Defined$ Player | Choices$ Discard,No | SubAbility$ DBSearch
+SVar:Discard:DB$ Discard | Defined$ Remembered | Mode$ TgtChoose | RememberDiscarded$ True | SpellDescription$ Discard a card.
```

Comment:
still not simultaneously :/

Reply:
why do you still loop instead of a single effect on Player.NotedForDiscard?

Reply:
Ugh misread the ruling, sorry